src/nrnpython/nrnpy_p2h.cpp

```diff
@@ -639,29 +639,25 @@ static char* pickle(PyObject* p, size_t* size) {
     }
     assert(r);
     assert(PyBytes_Check(r));
-    *size = PyBytes_Size(r);
-    char* buf1 = PyBytes_AsString(r);
-    char* buf = new char[*size];
-    for (size_t i = 0; i < *size; ++i) {
-        buf[i] = buf1[i];
-    }
+    std::size_t size = PyBytes_Size(r);
+    char* buf = PyBytes_AsString(r);
+    std::vector<char> ret(buf, buf + size);
     Py_XDECREF(r);
-    return buf;
+    return ret;
 }
 
-static char* po2pickle(Object* ho, std::size_t* size) {
+static std::vector<char> po2pickle(Object* ho) {
     setpickle();
     if (ho && ho->ctemplate->sym == nrnpy_pyobj_sym_) {
         PyObject* po = nrnpy_hoc2pyobject(ho);
-        char* buf = pickle(po, size);
-        return buf;
+        return pickle(po);
     } else {
-        return 0;
+        return {};
```

Comment:
We lost the ability to distinguish `nullptr` and `""`.

Reply:
I suspect my claim is incorrect, sensible serialization libraries need to store a version number of the format of the archive, otherwise they can't load old archives. On Python 3.6 (older than the oldest supported version) we have:
```
>>> pickle.dumps(None)
b'\x80\x03N.'
```
while on Python 3.11 we get:
```
>>> pickle.dumps(None)
b'\x80\x04N.'
```
which suggests that there's a header: `b'\x80\x04`.